### ISSUE TITLE
Some SSL flows not being detected due to record layer version assertion

### ISSUE THREAD

**Issue by dsokoloski** [ntop/nDPI#700]
I have noticed that the following line in ssl.c attempts to validate the second byte of a record version.  This can result in various SSL flows not being detected properly.

https://github.com/ntop/nDPI/blob/ea1ea99ce15885028c4d6aca84632961e37a20cf/src/lib/protocols/ssl.c#L762

According to RFC [5246](https://tools.ietf.org/html/rfc5246#appendix-E.1) Appendix E.1:

> Earlier versions of the TLS specification were not fully clear on
   what the record layer version number (TLSPlaintext.version) should
   contain when sending ClientHello (i.e., before it is known which
   version of the protocol will be employed).  Thus, TLS servers
   compliant with this specification MUST accept any value {03,**XX**} as
   the record layer version number for ClientHello.

In other words, it's possible here that any value could be in payload[2].  I have removed the above line in my test build and I am now able to correctly identify a lot more previously-guessed SSL flows -- especially those initiated from Windows workstations/servers.

I would normally submit a PR, but I think this should be reviewed by someone else more familiar with the protocol to validate my observation.

**Comment by utoni** [comment]
This issue is out-of-sync. ;)
